src/OGL3X/UniformBlock.cpp

```diff
@@ -148,7 +148,17 @@ void UniformBlock::setColorData(ColorUniforms _index, u32 _dataSize, const void
 		m_currentBuffer = m_colorsBlock.m_buffer;
 		glBindBuffer(GL_UNIFORM_BUFFER, m_colorsBlock.m_buffer);
 	}
-	glBufferSubData(GL_UNIFORM_BUFFER, m_colorsBlock.m_offsets[_index], _dataSize, _data);
+
+	OGLVideo & ogl = video();
+	OGLRender & render = ogl.getRender();
+	if (render.getRenderer() == OGLRender::glrAdreno)
+	{
+	        glBufferData(GL_UNIFORM_BUFFER, m_colorsBlockData.size(), m_colorsBlockData.data(), GL_STATIC_COPY);
```

Comment:
This alone fixes the flickering textures. glBufferSubData seems to be buggy in Adreno devices. I notice no performance loss after making this change.